src/ffi.rs

```diff
@@ -139,23 +139,23 @@ pub unsafe extern "C" fn free(ptr: *mut u8, cap: u32) {
 /// Get the public spend key with the given index.
 #[no_mangle]
 pub unsafe extern "C" fn public_spend_key(
-    index: u64,
+    index: &u64,
```

Comment:
After a long pairing (sorry for that) we came to the conclusion that we should use raw pointers here: not only for consistency but also because references need to be aligned and in FFI there is no guarantee about it when we pass a pointer.